### ISSUE TITLE
Bengali translation needed

### ISSUE THREAD

**Issue by BodhisattwaMandal** [eggpi/citationhunt#50]
Hi, can you get the tool in Bengali, so that we can use it for Bengali Wikipedia?

**Comment by eggpi** [comment]
Hey, I'll be happy to do that! I'll need a few bits of information on the Bengali Wikipedia:

- First of all, a complete translation of the interface to Bengali in [TranslateWiki]( https://translatewiki.net/w/i.php?title=Special:Translate&group=citationhunt&language=bn&filter=&action=translate). I see there is already a translation, but it's worth reviewing its quality.
- A link to an introductory article/tutorial about adding citations
- If there's a policy on omitting citations in the lead section (e.g., WP:CITELEAD for English), a link to that policy
- The name of a category containing all articles that need citations (it's fine if it has subcategories for years or months or so). I think [this](https://bn.wikipedia.org/wiki/%E0%A6%AC%E0%A6%BF%E0%A6%B7%E0%A6%AF%E0%A6%BC%E0%A6%B6%E0%A7%8D%E0%A6%B0%E0%A7%87%E0%A6%A3%E0%A7%80:%E0%A6%89%E0%A7%8E%E0%A6%B8%E0%A6%AC%E0%A6%BF%E0%A6%B9%E0%A7%80%E0%A6%A8_%E0%A6%A4%E0%A6%A5%E0%A7%8D%E0%A6%AF%E0%A6%B8%E0%A6%B9_%E0%A6%B8%E0%A6%95%E0%A6%B2_%E0%A6%A8%E0%A6%BF%E0%A6%AC%E0%A6%A8%E0%A7%8D%E0%A6%A7) would be the one for Bengali, but let me know if it's wrong.
- The names of templates that Citation Hunt should look for (for example, in French we look for {{Citation needed}}, {{Inédit}}, {{Refnec}} and a few others)

Once I have that information, it should be straightforward to do the translation. If you're interested in giving it a try, you can try adding a configuration for Bengali to [config.py](https://github.com/eggpi/citationhunt/blob/master/config.py) and following [these instructions](https://github.com/eggpi/citationhunt/blob/master/scripts/README.md#generating-the-database-locally) just to see if everything works fine, then sending me a pull request with the config.

Otherwise, I can do it myself later, once I have all the information I need :)

Thanks!

**Comment by BodhisattwaMandal** [comment]
Hi, thanks for your reply.
* Bengali translations of the interface in [Translatewiki](https://translatewiki.net/w/i.php?title=Special:Translate&group=citationhunt&language=bn&filter=&action=translate) have been reviewed completely. 
* An introductory tutorial about adding citations is [here](https://bn.wikipedia.org/s/1vcc), though not translated into local language, but we will do it on priority basis.
* WP:CITELEAD is also there in Bengali WP. 
* Yes, you are correct. [This ](https://bn.wikipedia.org/s/4ho) is the link of the category.
* Relevant templates [টেমপ্লেট:তথ্যসূত্র প্রয়োজন](https://bn.wikipedia.org/s/2awy), [টেমপ্লেট:তথ্যসূত্র যাচাই](https://bn.wikipedia.org/s/2lo5). Please let me know if any other links are needed.

Thanks.

**Comment by eggpi** [comment]
I set this up experimentally at https://tools.wmflabs.org/citationhunt/bn, please take a look. It's not yet linked to from any of the other languages. [Here's the commit](https://github.com/eggpi/citationhunt/commit/bbad3fed6295d4e92418a6df0b7ccc6af230d7fa) with the initial config.

Here's a few last questions:

- Are there are any alternative spellings of the templates you mentioned? For example, in English we explicitly check for {{cn}} and {{citation needed}} (this is case insensitive though, so {{Citation needed}} is covered). This will help us find more snippets.
- Do the snippets generally look readable, or are many of them broken? Our parser doesn't handle most Wikipedia templates, but I can work around that if a particularly common template is making lots of snippets look broken.
- Do the categories generally look useful? The algorithm for selecting them doesn't give me a lot of parameters to control how they're chosen, but I can tweak the total number of categories if needed - for example if they're too specific or broad.

I should really automate most of this configuration step soon, appreciate your patience. Looking forward for this to go live :)

**Comment by BodhisattwaMandal** [comment]
WP:CITELEAD redirects to Spanish Wikipedia instead of Bengali Wikipedia. The tool is showing old translatewiki translations, instead of recent translation. Thanks.

**Comment by NahidSultan** [comment]
Hello @eggpi 
We have changed some of the translations after seeing the beta version. Sorry for the confusion. The answer of your questions as follows:
* No, there is none but in case you need the name of their redirect page: [redirect pages](http://dispenser.homenet.org/~dispenser/cgi-bin/rdcheck.py?page=bn:%E0%A6%9F%E0%A7%87%E0%A6%AE%E0%A6%AA%E0%A7%8D%E0%A6%B2%E0%A7%87%E0%A6%9F:%E0%A6%A4%E0%A6%A5%E0%A7%8D%E0%A6%AF%E0%A6%B8%E0%A7%82%E0%A6%A4%E0%A7%8D%E0%A6%B0_%E0%A6%AA%E0%A7%8D%E0%A6%B0%E0%A6%AF%E0%A6%BC%E0%A7%8B%E0%A6%9C%E0%A6%A8) of টেমপ্লেট:তথ্যসূত্র প্রয়োজন and [redirect pages](http://dispenser.homenet.org/~dispenser/cgi-bin/rdcheck.py?page=bn:%E0%A6%9F%E0%A7%87%E0%A6%AE%E0%A6%AA%E0%A7%8D%E0%A6%B2%E0%A7%87%E0%A6%9F:%E0%A6%A4%E0%A6%A5%E0%A7%8D%E0%A6%AF%E0%A6%B8%E0%A7%82%E0%A6%A4%E0%A7%8D%E0%A6%B0_%E0%A6%AF%E0%A6%BE%E0%A6%9A%E0%A6%BE%E0%A6%87) of টেমপ্লেট:তথ্যসূত্র যাচাই.
* Yes, they are readable.
* and yes.
Looking forward to announce this new tool through bnwiki's social medias, VP and mailing list :)
Thanks.

**Comment by eggpi** [comment]
Sorry about the link to Spanish, should be fixed now. The translations are synchronized from TranslateWiki to this repository every once in a while, so I think I don't have the new ones yet. They should arrive in a couple of days though!

Thanks for the information, @NahidSultan! I've added the redirects to the config. I also filed #51 as this is something we should be able to handle automatically. But that's for another day.

I'll make the Bengali link visible from the other languages as soon as the translations are updated :)
